main/src/main/scala/sbt/internal/LibraryManagement.scala

```diff
@@ -19,6 +22,37 @@ private[sbt] object LibraryManagement {
 
   private type UpdateInputs = (Long, ModuleSettings, UpdateConfiguration)
 
+  def defaultUseCoursier: Boolean = {
+    val coursierOpt = sys.props
+      .get("sbt.coursier")
+      .flatMap(
+        str =>
+          ConsoleAppender.parseLogOption(str) match {
+            case LogOption.Always => Some(true)
+            case LogOption.Never  => Some(false)
+            case _                => None
+          }
+      )
+    val ivyOpt = sys.props
+      .get("sbt.ivy")
+      .flatMap(
+        str =>
+          ConsoleAppender.parseLogOption(str) match {
+            case LogOption.Always => Some(true)
+            case LogOption.Never  => Some(false)
+            case _                => None
+          }
+      )
+    val notIvyOpt = ivyOpt map { !_ }
+    coursierOpt.orElse(notIvyOpt).getOrElse(true)
+  }
+
+  def dependencyResolutionTask: Def.Initialize[Task[DependencyResolution]] = Def.taskDyn {
```

Comment:
Does this means that changing the from Ivy to Coursier in the shell does work already?

Reply:
It should.